vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -0,0 +1,30 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OrderHandler
+
+      def self.process(occupancy)
+        return unless occupancy.orderable?
+
+        order = Order.create!(
+          account: occupancy.account,
+          user: occupancy.user,
+          facility: occupancy.facility,
+          created_by_user: occupancy.user,
+        )
+
+        order.order_details.create!(
+          product: occupancy.secure_room,
+          created_by_user: occupancy.user,
+          quantity: 1,
```

Comment:
Also that `OrderDetail#state == "new"` (or `inproces`). There is both a state and a status on order details. The states are new->inprocess->complete->reconciled (also canceled). The statuses are those, plus there are sub-statuses (e.g. "Pending Approval" is a sub-status of "New"). Order details have a `new` state from their creation, but don't get an `order_status` until they're purchased.